Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -1498,11 +1516,40 @@ public string GetLocalApiUrl(IpAddressInfo ipAddress)
 
         public string GetLocalApiUrl(string host)
         {
+            if (EnableHttps)
+            {
+                return string.Format("https://{0}:{1}",
+                    host,
+                    HttpsPort.ToString(CultureInfo.InvariantCulture));
+            }
             return string.Format("http://{0}:{1}",
-                host,
-                HttpPort.ToString(CultureInfo.InvariantCulture));
+                    host,
+                    HttpPort.ToString(CultureInfo.InvariantCulture));      
         }
 
+        public string GetWanApiUrl(IpAddressInfo ipAddress)
+        {
+            if (ipAddress.AddressFamily == IpAddressFamily.InterNetworkV6)
+            {
+                return GetWanApiUrl("[" + ipAddress.Address + "]");
+            }
+
+            return GetWanApiUrl(ipAddress.Address);
+        }
+
+        public string GetWanApiUrl(string host)
+        {
+            if (EnableHttps)
+            {
+                return string.Format("https://{0}:{1}",
+                    host,
+                    ServerConfigurationManager.Configuration.PublicHttpsPort.ToString(CultureInfo.InvariantCulture));
+            }
+            return string.Format("http://{0}:{1}",
```

Comment:
```suggestion

            return string.Format("http://{0}:{1}",
```
Empty line after closing brace.